ci-operator/step-registry/telcov10n/virtualised-hub-cluster/tests/telcov10n-virtualised-hub-cluster-tests-commands.sh

```diff
@@ -0,0 +1,534 @@
+    ("$(oc whoami --show-console)", 200),
+    ("$(oc --kubeconfig ${ARTIFACT_DIR}/kubeconfig whoami --show-console)", 200),
+    ("$(oc get managedcluster local-cluster -ojsonpath='{.spec.managedClusterClientConfigs[0].url}')", 403),
+])
+def test_http_endpoint(url):
+    response = requests.get(url[0], verify=False)
+    assert response.status_code == url[1], f"Endpoint {url[0]} is not accessible. Status code: {response.status_code}"
+
+def test_oc_commands(bash):
+
+    node_list=[
+      "node/openshift-master-0.${CLUSTER_DOMAIN}",
+      "node/openshift-master-1.${CLUSTER_DOMAIN}",
+      "node/openshift-master-2.${CLUSTER_DOMAIN}",
+    ]
+    with bash(envvars={'KUBECONFIG': '${ARTIFACT_DIR}/kubeconfig'}) as s:
+      assert s.run_script_inline(['oc get no -oname']) == "\n".join(node_list)
+
+    assert bash.run_script_inline(['oc get no -oname']) == "\n".join(node_list)
+
+def test_cluster_version(bash):
+
+    oc_cmd = "oc get clusterversion version -ojsonpath='{.status.desired.version}'"
+    assert bash.run_script_inline([oc_cmd]).startswith("${CLUSTER_TAG}")
+
+@pytest.mark.parametrize("namespace", [
+    "openshift-gitops",
+    "open-cluster-management",
+    "openshift-local-storage",
+    "multicluster-engine",
+])
+def test_ztp_namespaces(bash, namespace):
+    oc_cmd = f"oc get ns {namespace} " + "-ojsonpath='{.metadata.name}'"
+    assert bash.run_script_inline([oc_cmd]) == namespace
+
+    count = 0
+    attempts = 10
+    while attempts > 0:
+      oc_cmd = f"oc -n {namespace} get po --no-headers | grep -v -E 'Running|Completed' | wc -l"
+      if bash.run_script_inline([oc_cmd]) == '0':
+        break
+      attempts -= 1
+      time.sleep(60)
+    assert attempts > 0, f"Not all PODs in {namespace} namespace are ready yet"
+
+# @pytest.mark.parametrize("gitea", [
+#   {"url": "${GITEA_URL}", "status_code": 200},
+# ])
+# def test_gitea_service(gitea):
+#     response = requests.get(gitea['url'], verify=False)
+#     assert response.status_code == gitea['status_code'], f"Endpoint {gitea['url']} is not accessible. Status code: {response.status_code}"
+EOF
+
+  run_pytest check_hub_installation
+}
+
+##############################################################################################################
+# Clean up
+##############################################################################################################
+
+function clean_up {
+
+  echo "************ telcov10n-vhub Clean up ************"
+
+  run_cmd kcli delete vm openshift-bootstrap --yes || echo
+}
+
+##############################################################################################################
+# Main Entry point
+##############################################################################################################
+
+function main {
+  # install_deps
+  load_settings
+  setup_remote_hipervisor
+  deploy_virtualised_hub_cluster
+  oc_login
+  setup_gitea
+  gather_info
+  test_deployment_and_services
```

Comment:
Maybe we should split this into multiple steps? so we will have pre/post steps and tests